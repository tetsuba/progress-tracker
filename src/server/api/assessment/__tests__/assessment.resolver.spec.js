const { createTestServer } = require('../../../test/serverTestHelper')
const {
  StudentMockRequest,
} = require('../../../../test/mockApi/student/studentMockData')
const {
  AlphabetMockRequest,
} = require('../../../../test/mockApi/assessment/alphabet/alphabetMockData')
const {
  ADD_ASSESSMENT_MUTATION,
} = require('../../../../client/api/assessment/assessment.mutation')

describe('assessment.resolver', () => {
  describe('@Mutation', () => {
    describe('addAssessment', () => {
      const ctx = {
        models: {
          Student: new StudentMockRequest(),
          AlphabetAssessment: new AlphabetMockRequest(),
        },
      }

      async function addAssessmentTest(assessmentName, studentId) {
        const { mutate } = createTestServer(ctx)
        return await mutate({
          mutation: ADD_ASSESSMENT_MUTATION,
          variables: {
            input: {
              studentId: studentId,
              assessmentName: assessmentName,
            },
          },
        })
      }

      describe('Success', () => {
        describe('Add new assessment to student assessments', () => {
          const studentId = 'studentId01'
          test('should add assessment "alphabet-upperCase"', async () => {
            const res = await addAssessmentTest('alphabet-upperCase', studentId)
            expect(res).toMatchSnapshot()
          })
          test('should add assessment "alphabet-lowerCase"', async () => {
            const res = await addAssessmentTest('alphabet-lowerCase', studentId)
            expect(res).toMatchSnapshot()
          })
          test('should add assessment "alphabet-name"', async () => {
            const res = await addAssessmentTest('alphabet-name', studentId)
            expect(res).toMatchSnapshot()
          })
          test('should add assessment "alphabet-sound"', async () => {
            const res = await addAssessmentTest('alphabet-sound', studentId)
            expect(res).toMatchSnapshot()
          })
        })
      })
      describe('Error', () => {
        test('should return an error if student not found', async () => {
          const studentId = 'incorrect-student-id'
          const res = await addAssessmentTest('alphabet-lower', studentId)
          expect(res.errors).toMatchSnapshot()
        })
      })
    })
  })
})
