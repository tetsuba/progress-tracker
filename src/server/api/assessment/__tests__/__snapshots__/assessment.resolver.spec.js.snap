// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assessment.resolver @Mutation addAssessment Error should return an error if student not found 1`] = `
Array [
  [GraphQLError: STUDENT_NOT_FOUND],
]
`;

exports[`assessment.resolver @Mutation addAssessment Success Add new assessment to student assessments should add assessment "alphabet-lowerCase" 1`] = `
Object {
  "data": Object {
    "addAssessment": Object {
      "assessments": Array [
        "Assessment",
        "Assessment-1",
        "alphabet-lowerCase",
      ],
      "firstName": "David",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`assessment.resolver @Mutation addAssessment Success Add new assessment to student assessments should add assessment "alphabet-name" 1`] = `
Object {
  "data": Object {
    "addAssessment": Object {
      "assessments": Array [
        "Assessment",
        "Assessment-1",
        "alphabet-name",
      ],
      "firstName": "David",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`assessment.resolver @Mutation addAssessment Success Add new assessment to student assessments should add assessment "alphabet-sound" 1`] = `
Object {
  "data": Object {
    "addAssessment": Object {
      "assessments": Array [
        "Assessment",
        "Assessment-1",
        "alphabet-sound",
      ],
      "firstName": "David",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`assessment.resolver @Mutation addAssessment Success Add new assessment to student assessments should add assessment "alphabet-upperCase" 1`] = `
Object {
  "data": Object {
    "addAssessment": Object {
      "assessments": Array [
        "Assessment",
        "Assessment-1",
        "alphabet-upperCase",
      ],
      "firstName": "David",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
